server/src/main/java/org/opensearch/cluster/routing/allocation/decider/ThrottlingAllocationDecider.java

```diff
@@ -150,36 +165,75 @@ public Decision canAllocate(ShardRouting shardRouting, RoutingNode node, Routing
             // Peer recovery
             assert initializingShard(shardRouting, node.nodeId()).recoverySource().getType() == RecoverySource.Type.PEER;
 
-            // Allocating a shard to this node will increase the incoming recoveries
-            int currentInRecoveries = allocation.routingNodes().getIncomingRecoveries(node.nodeId());
-            if (currentInRecoveries >= concurrentIncomingRecoveries) {
+            if(shardRouting.unassignedReasonIndexCreated()) {
+                return allocateInitialReplicas(shardRouting, node, allocation);
+            } else {
+                return allocateNonIntialReplicas(shardRouting, node, allocation);
+            }
+        }
+    }
+
+    private Decision allocateInitialReplicas(ShardRouting shardRouting, RoutingNode node, RoutingAllocation allocation) {
+
+        int currentInRecoveries = allocation.routingNodes().getInitialIncomingRecoveries(node.nodeId());
+
+        return allocateReplicas(shardRouting, allocation, currentInRecoveries, replicasInitialRecoveries,
+            (x,y) -> getInitialPrimaryNodeOutgoingRecoveries(x,y), replicasInitialRecoveries,
+            String.format(Locale.ROOT, "[%s=%d]", CLUSTER_ROUTING_ALLOCATION_NODE_CONCURRENT_INCOMING_RECOVERIES_SETTING.getKey(),
```

Comment:
here in the message it shouldnt it be - CLUSTER_ROUTING_ALLOCATION_NODE_INITIAL_REPLICAS_RECOVERIES_SETTING this setting as initial replica recoveries are counted towards this?